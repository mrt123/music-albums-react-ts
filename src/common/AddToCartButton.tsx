const buttonStyle = {
  background: "rgb(237, 92, 145)",
  height: "40px",
  cursor: "pointer",
};

const AddToCartButton = () => {
  return <button style={buttonStyle}>Add</button>;
};

export default AddToCartButton;
