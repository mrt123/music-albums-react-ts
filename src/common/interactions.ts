import { fireEvent } from "@testing-library/react";

export const click = (element: HTMLElement) => {
  fireEvent(
    element,
    new MouseEvent("click", {
      bubbles: true,
      cancelable: true,
    })
  );
};

export const clickNTimes = (element: HTMLElement, n: number) => {
  for (var i = 0; i < n; i++) {
    click(element);
  }
};
