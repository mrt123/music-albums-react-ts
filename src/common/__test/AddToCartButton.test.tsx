import { render } from "@testing-library/react";
import AddToCartButton from "../AddToCartButton";
import { click } from "../interactions";

describe("AddToCartButton.tsx", () => {
  it("matches spanshot", () => {
    const { container } = render(<AddToCartButton onClick={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  it("expects onClick function to be called (sposob pierwszy)", () => {
    let isExecuted = false;
    const zmienZmienna = () => {
      isExecuted = true;
    };
    const { getByTestId } = render(<AddToCartButton onClick={zmienZmienna} />);

    const addButtonElement = getByTestId("addToCartButton");
    click(addButtonElement);
    expect(isExecuted).toEqual(true);
  });

  it("expects onClick function to be called (sposob 2)", () => {
    const executeOnclick = jest.fn();

    const { getByTestId } = render(
      <AddToCartButton onClick={executeOnclick} />
    );

    const addButtonElement = getByTestId("addToCartButton");
    click(addButtonElement);
    click(addButtonElement);
    expect(executeOnclick).toHaveBeenCalledTimes(2);
  });
});
