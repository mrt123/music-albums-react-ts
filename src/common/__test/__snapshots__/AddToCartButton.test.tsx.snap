// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddToCartButton.tsx matches spanshot 1`] = `
<div>
  <button
    data-testid="addToCartButton"
    style="background: rgb(237, 92, 145); cursor: pointer; border-radius: 4px; padding: 12px;"
  >
    Add
  </button>
</div>
`;
